core/storage/translation/gae_models.py

```diff
@@ -34,6 +35,160 @@
 datastore_services = models.Registry.import_datastore_services()
 
 
+class EntityTranslationsModel(base_models.BaseModel):
+    """Model for storing entity translations."""
+
+    # The id of the corresponding entity.
+    entity_id = datastore_services.StringProperty(required=True, indexed=True)
+    # The type of the corresponding entity.
+    entity_type = datastore_services.StringProperty(
+        required=True, indexed=True, choices=[
+            feconf.ENTITY_TYPE_EXPLORATION,
+            feconf.ENTITY_TYPE_QUESTION
+        ])
+    # The version of the corresponding entity.
+    entity_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The ISO 639-1 code for the language an entity is written in.
+    language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # A dict representing content-id as keys and dict(TranslatedContent)
+    # as values.
+    translations = datastore_services.JsonProperty(required=True)
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model does not contain user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'entity_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'entity_type': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'entity_version': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'language_code': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'translations': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @staticmethod
+    def _generate_id(
+        entity_type: feconf.TranslatableEntityType,
+        entity_id: str,
+        entity_version: int,
+        language_code: str
+    ) -> str:
+        """The method use to generate unique id for entity translations model.
+
+        Args:
+            entity_type: TranslatableEntityType. Type
+                of the entity to fetch.
+            entity_id: str. Id of an entity.
+            entity_version: int. Version of an entity.
```

Comment:
The ID of the entity.

The version of the entity.

The language code for the entity.

Reply:
Updated.